src/setupProxy.js

```diff
@@ -1,40 +1,96 @@
 const { createProxyMiddleware } = require('http-proxy-middleware')
 
-const { PUBLIC_URL = '' } = process.env
+const { PUBLIC_URL = '', JAM_TARGET_CONTAINER = 'PRIMARY' } = process.env
 
-module.exports = (app) => {
-  app.use(
-    createProxyMiddleware(`${PUBLIC_URL}/api/`, {
-      target: 'https://localhost:28183',
-      pathRewrite: { [`^${PUBLIC_URL}`]: '' },
-      changeOrigin: true,
-      secure: false,
-      ws: false,
-      onProxyReq: (proxyReq, req, res) => {
-        if (req.headers['x-jm-authorization']) {
-          proxyReq.setHeader('Authorization', req.headers['x-jm-authorization'])
-        }
-      },
-    })
-  )
+const PRIMARY_CONTAINER = {
+  apiPort: 28183,
+  websocketPort: 28283,
+  obwatcherPort: 62601,
+}
 
-  // https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/JSON-RPC-API-using-jmwalletd.md#websocket
-  app.use(
-    createProxyMiddleware(`${PUBLIC_URL}/jmws`, {
-      target: 'https://localhost:28283',
-      pathRewrite: { [`^${PUBLIC_URL}/jmws`]: '' },
-      changeOrigin: true,
-      secure: false,
-      ws: true,
-    })
-  )
+const SECONDARY_CONTAINER_JAM_API_PORT = 29080
```

Comment:
Not sure about the `PRIMARY` & `SECONDARY` naming. Shouldn't it be `standalone` and `ui-only`?

The primary/secondary distinction is for our regtest setup only, or is it not?

Reply:
You've hit a real pain point.

First off, I totally agree - `ui-only` and `standalone` is already way better naming.
`primary` and `secondary` is just too tightly coupled to the regtest env and the docs mention `npm start` as an alternative to running the docker images.

However, `ui-only` and `standalone` are the terminologies to different the docker distributions.
In this case, "secondary" is really using a `standalone` docker image as backend. But "primary" does not use `ui-only`, it just "mimics" it with `setupProxy`. This might lead to confusion (at least it is already confusing to me to some extent).

Just for the record:
Imho, the correct approach would be, instead of advocating using `npm start` to access a mainnet jm instance,
building Jam (with `npm build`) and delivering it via a real web server. But that seems overly complicated for "ordinary" users - that's why the `ui-only` docker image is so handy - it does this out-of-the-box.

Having said that - it is nice that `npm start` also works as a quick way to start Jam and we should aim to keep it that way.

Thanks for your input :orange_heart:
I'll try to come up with a more suitable approach.